crates/risc-0/methods/guest/Cargo.toml

```diff
@@ -0,0 +1,19 @@
+[package]
+name = "zk-aquavm"
+version = "0.1.0"
+description = "RISC-0 guest wrapper for AquaVM"
+authors = ["Fluence Labs"]
+edition = "2021"
+license = "Apache-2.0"
+publish = false
+
+[workspace]
+
+[dependencies]
+risc0-zkvm = { version = "0.19.1", default-features = false, features = [ "std" ] }
+zk-aquavm-interface = { path = "../../core" }
+aquavm-air = { path = "../../../../air", version = "0.55.0" }
+
+[patch.crates-io]
+fluence-keypair = { git = "https://github.com/fluencelabs/trust-graph.git", branch = "lean-keypair-2" }
+libp2p-identity = { git = "https://github.com/fluencelabs/rust-libp2p.git", branch = "rand-feature" }
```

Comment:
I believe it is already in the libp2p master.  And the fluence-keypair was an appropriate fix too.

Reply:
It is, but there will be another problem b/c Risc-0 supports only Rust 1.69-dev which is too old for libp2p:
```
zk-aquavm: error: package `libp2p-identity v0.2.8` cannot be built because it requires rustc 1.73.0 or newer, while the currently active rustc version is 1.69.0-dev           
z
```

Reply:
0.2.7 should work:
https://crates.io/crates/libp2p-identity/versions
https://docs.rs/crate/libp2p-identity/0.2.7/features

Reply:
There are two problems:
 - as I understand it's not possible to disable or enable features in the patch section if you point to the same version, Rust will try to enable all features that are enabled in all transitive deps of this crate. 
 - currently there is no way in patch to specify a different version than the original dependency, one can only specify the different repo or local path

Reply:
For the second an error will be as follows:
```
zk-aquavm: Caused by:
zk-aquavm:   patch for `libp2p-identity` in `https://github.com/rust-lang/crates.io-index` points to the same source, but patches must point to different sources

```

Reply:
How should it work instead? There is no `libp2p-identity` dep anywhere in the AquaVM repo.

Reply:
I don't know how to verify correctly, but I removed the patch section and added
```
libp2p-identity = { version = "=0.2.7", default-features = false }
```
to the dependencies, and the `cargo build` worked.

Reply:
Hmm, I understand how version pinning worked here, but I don't think that disabling default-features will really work here b/c the Rust compiler will use all enabled features and won't allow to turned them off...

Reply:
We decided to skip this off until RISC-0 supports the newest version of Rust which will make it compilable with the `libp2p-identity` 0.2.8 and then we can implement the `rand` syscall.